Upgraded SSP tutorial - Aligned with the new implementation of a node…

joeiacono2021 opened the pull request:
…mon service provider - pre review

Still need to add the correct url to css file (to be hosted on github) and will be done when  updated tutorial code is merged in

joeiacono2021 committed:
Upgraded SSP tutorial - Aligned with the new implementation of a nodemon service provider - pre review

Still need to add the correct url to css file (to be hosted on github) and will be done when  updated tutorial code is merged in

mfahampshire commented on src/tutorials/simple-service-provider.md:
```diff
@@ -8,8 +8,8 @@ In this guide, developers will learn how to build the foundation of a privacy-en
+- A __Service Provider__ also written in TypeScript, which can receive messages from the mixnet.
 - Additionally, you will be guided on how to configure a pair of Nym Websocket Clients, which are necessary for connecting to the mixnet with your application.
 
 > ⚠️ Service providers are usually run on remote servers to keep metadata private, but for demonstration purposes, this tutorial will show how to run it on a local machine using looped messages through the mixnet.
```
I get what you mean with 'using looped messages through the mixnet' but just change it to 'sending messages to ourselves' or something to be more clear

mfahampshire commented on src/tutorials/simple-service-provider.md:
```diff
@@ -138,21 +141,20 @@ The package (`ts-node`) allows us to build a typescript application in a node en
+To build and run our application locally, we require a tool that allows us to work on it while it's running and instantly reflects saved changes on the browser.
 
-1. This can be achieved through the installation of Parcel bundler using the following command in your terminal window:
+1.  This can be achieved through the installation of Parcel bundler using the following command in your terminal window:
```
Link to parcel

mfahampshire replied:
Do we need `parcel` installed globally? I don't know whether we should be requiring global installs necessarily

mfahampshire commented on src/tutorials/simple-service-provider.md:
```diff
@@ -176,34 +178,34 @@ After completing the installation process, start by creating a `src` folder with
     
 ### Building the User Client
 
 Building our __User Client__ focuses mainly on configuring and adding functions to `index.ts` file, in the `user-client` folder. It's where we establish the connection between our application and the mixnet, create the messages we want to send, and then send them to a service provider through the mixnet. 
```
shouldn't it be 'send them to our service provider through the mixnet'?

mfahampshire commented on src/tutorials/simple-service-provider.md:
````diff
@@ -223,394 +225,415 @@ Therefore we must implement the functions that connects our Typescript Client to
     main();
     ```
-* `main()` - This first function will hold the majority of the logic and initiate the connection with the Nym Websocket Client. It's best to declare it at the start of the file and call it at the end to run when launching the application.
+    * `main()` - This first function will hold the majority of the logic and initiate the connection with the Nym Websocket Client. It's best to declare it at the start of the file and call it at the end to run when launching the application.
````
You don't need to call it a 'Nym Websocket Client' each time - just 'websocket client' is fine after the first introduction of it imo

mfahampshire commented on src/tutorials/simple-service-provider.md:
````diff
@@ -223,394 +225,415 @@ Therefore we must implement the functions that connects our Typescript Client to
+    ```
 
-The majority of this code is required for adjusting the elements of the `index.html`. 
+    The majority of this code is required for adjusting the elements of the `index.html`. 
````
Can you rewrite this sentence? Not 100% sure what it means, needs to be clearer

mfahampshire commented on src/tutorials/simple-service-provider.md:
```diff
@@ -223,394 +225,415 @@ Therefore we must implement the functions that connects our Typescript Client to
+                type: "send",
+                message: JSON.stringify(messageContentToSend),
+                recipient: targetAddress,
+                withReplySurb: false,
```
This message format is now out of date - `withReplySurb` no longer exists. Also make it clear that we're sending this message with a cleartext 'reply to' address for the sake of simplicity, whereas in prod you'd use SURBs. 

cf. https://nymtech.net/docs/clients/websocket-client.html#message-types

mfahampshire commented on src/tutorials/simple-service-provider.md:
````diff
@@ -223,394 +225,415 @@ Therefore we must implement the functions that connects our Typescript Client to
-```
+    ```
 
-* `sendMessageToMixnet()` - The key function that will allow our Service Provider messages to receive messages. Firstly, it will gets the values from a form in the `index.html` and assign them to local variables within the function, inserting the local variables into one object to be sent to the mixnet. Secondly, calling the `displayJsonSend()` function to render the sent message on to the UI. La ...
````
'it will gets' needs correcting. Overall the tense jumps around a lot in the text - pick one and stick to it throughout please.

mfahampshire commented on src/tutorials/simple-service-provider.md:
```diff
@@ -223,394 +225,415 @@ Therefore we must implement the functions that connects our Typescript Client to
+    Lets add the finishing touches to the UI by adding in the stylesheet which we specified at the top our our `index.html` above. Back in our project root folder (same folder level as `/src`), create an `/assets` folder and create a new file within it called `styles.css`. 
 
-9. Return back to your terminal and run:
+    We have made a stylesheet for this tutorial which can be accessed [here](Link to be added when merged). Copy and paste the contents into your newly created `styles.css` file, or simply paste in your own stylesheet! Once we've checked our browser to confirm our styles are working, lets move onto getting our application running.
```
If we're going through the trouble of manually adding a stylesheet instead of importing it from a CDN, why are we doing this? Add a little explainer

mfahampshire commented on src/tutorials/simple-service-provider.md:
```diff
@@ -223,394 +225,415 @@ Therefore we must implement the functions that connects our Typescript Client to
+        Using gateway: 5Ao1J38frnU9Rx5YVeF5BWExcnDTcW8etNe9W2sRASXD
+        Client configuration completed.
+
+        Version: 1.1.4
```
Actually run the code on a client of the correct version to get the console output please. Pretty sure the logging has changed since 1.1.4

mfahampshire replied:
this goes for all the console output in the doc

mfahampshire commented on src/tutorials/simple-service-provider.md:
```diff
@@ -223,394 +225,415 @@ Therefore we must implement the functions that connects our Typescript Client to
+    </details>
 
-> ⚠️ The client address generated by executing a command in a terminal will always be unique and distinct from the address generated by any other client executing the same command.
+    > ⚠️ The client address generated by executing a command in a terminal will always be unique and distinct from the address generated by any other client executing the same command.
```
simplify this sentence please

mfahampshire commented on src/tutorials/simple-service-provider.md:
```diff
@@ -223,394 +225,415 @@ Therefore we must implement the functions that connects our Typescript Client to
-From the [Preparing your Typescript environment](#preparing-your-typescript-environment) section of tutorial guide, repeat the steps `1-4` and step `1` of [Bundling the Application](#bundling-the-application) for adding the __Service Provider__.
+From the [Preparing your Typescript environment](#preparing-your-typescript-environment) section of tutorial guide, repeat the steps `1` and `2` for adding the __Service Provider__.
 
 > ⚠️ Remember to reference this section of the tutorial to the `service-provider`, instead of `user-client`. 
```
reword - just say that they should repeat the steps from above but calling the directory `service-provider` instead of `user-client`

mfahampshire commented on src/tutorials/simple-service-provider.md:
````diff
@@ -223,394 +225,415 @@ Therefore we must implement the functions that connects our Typescript Client to
 
-By stating the `--port 1235`, it ensures it runs on `localhost:1235` instead of the its User Clients `localhost:1234`, to prevent any conflicting errors.
+    ```
+    For our __Service Provider__ , we will be running it as a `nodemon` application that will run in the terminal.
````
Would we do this in prod? If not, why are we doing it here like this now? Add info

mfahampshire commented on src/tutorials/simple-service-provider.md:
```diff
@@ -223,394 +225,415 @@ Therefore we must implement the functions that connects our Typescript Client to
-         | | | | |_| | | | | | |
-         |_| |_|\__, |_| |_| |_|
-                |___/
+This far into the tutorial, we should have functioning __User Client__ to make the initial websocket connection that we're looking for. To connect our Nym Websocket client, go to [releases page](https://github.com/nymtech/nym/releases) to download the latest binaries release of the `nym-client`. Alternatively, download [here](https://nymtech.net/docs/stable/run-nym-nodes/build-nym/) and follow in ...
```
Reword this paragraph. 

If you follow the first link you just download one binary release (make a note of which OS the binary release is for)

The second link is not going to a download page so I'm not sure why 'download' is mentioned there.

mfahampshire commented on src/tutorials/simple-service-provider.md:
```diff
@@ -724,163 +718,70 @@ Add the following block of functions and variable into the `index.ts`:
-            <div class="ui list">
-            <div id="slideContainer"></div>
-            </div>
+> You may observe that we possess similar functions to the __User Client__ code, with only a few differences in the objectives of a few functions, mainly our `handleResponse()` function.
```
Reword this sentence please, doesn't really make sense to talk of the 'objectives of a few functions'

mfahampshire commented on src/tutorials/simple-service-provider.md:
```diff
@@ -724,163 +718,70 @@ Add the following block of functions and variable into the `index.ts`:
+> You may observe that we possess similar functions to the __User Client__ code, with only a few differences in the objectives of a few functions, mainly our `handleResponse()` function.
 
-            <hr>
+* `main()` - Just like our __User Client__, our `main()` function will still be the function in charge of our initializing and executing our application. We connect to our websocket in the exact same way as we do in our __User Client__ code except we want to set our `port` local variable to '1978'. This is so we don't have a conflict with the other Nym Websocket Client (the one that we are runnin ...
```
Needs a reword, the grammar of the last sentence is off - seems to be a run on from the previous sentence, but is in fact a new one?

mfahampshire commented on src/tutorials/simple-service-provider.md:
```diff
@@ -724,163 +718,70 @@ Add the following block of functions and variable into the `index.ts`:
-            </div>
-            
-            <h3 style="margin-left:10px">Activity Log</h3>
+* `handleResponse()` - When it comes to the sorting between types of incoming messages, its works relatively the same as the function with the same name defined in our __User Client__. The main difference is within how we are logging our data (which will be displayed in our terminal once the application is up and running).
```
grammar fixes needed

mfahampshire commented on src/tutorials/simple-service-provider.md:
````diff
@@ -724,163 +718,70 @@ Add the following block of functions and variable into the `index.ts`:
-        </body>
-    </html>
-```
+The '\x1b' orefix you see in our console.log enables the ability for us to color our 'console.log' statements. The number that you see following the '[' and preceeding 'm' is the color code that can be compared here - https://en.m.wikipedia.org/wiki/ANSI_escape_code#Colors. Its a nice and quick way of styling our terminal output.
````
spelling and grammar fixes here - use british english spellings throughout as well please, not american english

mfahampshire commented on src/tutorials/simple-service-provider.md:
````diff
@@ -724,163 +718,70 @@ Add the following block of functions and variable into the `index.ts`:
-    │  ├─ package.json
-    │  ├─ tsconfig.json
-```
+Before we get our __Service Provider__ up and running, lets get our second NYm WEbsocket Client up and running
````
capitalisation errors

mfahampshire commented on src/tutorials/simple-service-provider.md:
````diff
@@ -724,163 +718,70 @@ Add the following block of functions and variable into the `index.ts`:
-```
-var targetAddress = '<service-provider-websocket-client-addresss>';
-```
+    We can see that we get our Nym Websocket Address successfully.
````
its not a websocket address, its a client address!

mfahampshire commented on src/tutorials/simple-service-provider.md:
````diff
@@ -724,163 +718,70 @@ Add the following block of functions and variable into the `index.ts`:
+    var targetAddress = '<service-provider-websocket-client-addresss>';
+    ```
 
+    > ⚠️ Reminder that the client address produced from running this command in your terminal will always be different to the one you have when you execute this command. Each address generated by each client will be different.
````
Why will it be different? What makes up a client address? Link to the docs + do grammar fixes

mfahampshire commented on src/tutorials/simple-service-provider.md:
```diff
@@ -724,163 +718,70 @@ Add the following block of functions and variable into the `index.ts`:
 
-* A Single 'User Client' Web App running in the terminal
-* A Single 'Service Provider' Web App running in the terminal 
+* A Single __User Client__  Web App running in the terminal and its UI on the Web Browser
```
'in' not 'on' a web browser + fix capitalisation

mfahampshire commented on src/tutorials/simple-service-provider.md:
```diff
@@ -724,163 +718,70 @@ Add the following block of functions and variable into the `index.ts`:
+We can now attempt to send a message by completing the fields on the __User Client__ browser app and pressing __'Send'__.
 
-After clicking __'Send'__, you can observe a message sent from the 'User Client' to the 'Service Provider' via the mixnet in the Service Provider browser app, thus creating a basic web application solution in the process.
+After clicking __'Send'__, you can observe a message sent from the __User Client__ to the __Service Provider__ via the mixnet in the browser app, thus creating a basic web application solution in the process.
```
Needs a better concluding sentence

mfahampshire reviewed (changes requested):
Additional changes requested: there are still references to a service provider UI that no longer exists - remove please

joeiacono2021 committed:
PR updates v1. Ammendments in description

- Changed the occurance of Nym Websocket Client to `nym-client`
- Added explainers of UI based manipulation functions
- Capitalization and typo corrections
- Pre Parcel --global check

joeiacono2021 committed:
Corrected the initialization stage when creating our service provider .

- Only repeat step 1 of the User Client instructions.

mfahampshire committed:
temp commit

mfahampshire committed:
* broke tutorial down into sub chapters
* added sub chapters to summary.md

mfahampshire committed:
added new directory structure

mfahampshire committed:
finished first rewrite

mfahampshire commented:
todo: 
- [x] redraw diagram in ascii 
- [x] make sure subheadings are `##` not `###` in https://dev-portal-feature-ssp-tutorial-rewrite.ci.nymte.ch/tutorials/simple-service-provider/preparating-env.html#prerequisites
- [x] fix formatting + make sure all subheadings are `##`  https://dev-portal-feature-ssp-tutorial-rewrite.ci.nymte.ch/tutorials/simple-service-provider/user-client.html#connecting-to-your-nym-client
- [x] add stylesheet link once code repo PR is merged 
- [x] new screenshots

mfahampshire committed:
small formatting tweaks

mfahampshire committed:
fixed incorrect mkdir command

mfahampshire committed:
fixed typo

mfahampshire committed:
fixed typo

mfahampshire committed:
finalising content rewrite

mfahampshire committed:
replaced diagram with ascii

mfahampshire committed:
added curl command to grab stylesheet

mfahampshire committed:
replaced screenshots with surb-enabled ones
